add documentation for PQS (#457)
digital-asset/docs.daml.com: docs/2.7.1/docs/query/pqs-user-guide.rst

  event_offset >= oldest_offset

Write Pipeline
--------------

Only advanced users should be concerned with the manner in which the
write pipeline is implemented. The above Read API takes into
consideration the manner in which the write pipeline is implemented, and
therefore the above Read API is the recommended way to query the
datastore. However, for completeness we provide the following
information.

A Daml transaction is a collection of events that take effect on the
ledger atomically. However it needs to be noted that for performance
reasons, these transactions are written to the datastore *in parallel*,
and whilst the datastore is written to in a purely append-only fashion,
it is not guaranteed that these transactions will become visible to
readers in order. The offset-based model makes the database’s isolation
level irrelevant - so the loosest model (``read uncommitted``) is not
harmful.

The first thing to consider when querying the datastore is the type of
read consistency required. If there is no need for consistency (eg.
reading a historical contract - regardless of lifetime) then payload
tables can be queried directly, without any consideration of offset.

> Another example could be a liveness metric query that calculates the

Reviewer: ```suggestion
Another example is a liveness metric query that calculates the
```

number of transactions in the datastore over the past minute. Again,
this could be entirely valid without consideration of the
parallel-writing method.

When consistency is required, the reader must be aware of the offset

> from which they are reading. This will ensure they do not also read

Reviewer: ```suggestion
from which they are reading. This ensures they do not also read
```

> further offsets that are present - but their precedent events are not yet

Reviewer: ```suggestion
further offsets that are present, but their precedent events are not yet
```
